### PR TITLE
ISPN-8935 AbstractFileLookup.lookupFileStrict does not work for jar scheme

### PR DESCRIPTION
**Pull request by dmurat**

### CODE REVIEW

**Comment by tristantarrant** [comment]
ok to test

**Comment by tristantarrant** [comment]
test this please

File: jcache/embedded/src/test/java/org/infinispan/jcache/JCacheConfigurationTest.java [hunk 1]
```diff
@@ -40,18 +53,96 @@ public void testJCacheManagerWherePathContainsFileSchemaAndAbsolutePath() throws
...
+         if (originalClassLoader != null) {
+            Thread.currentThread().setContextClassLoader(originalClassLoader);
+         }
+      }
+   }
+
+   private void createJar(File jarFile, String existingResourceToAddInJar, String targetPathInJar) throws IOException, URISyntaxException {
+      Manifest manifest = new Manifest();
+      manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+      JarOutputStream sampleJarOutputStream = new JarOutputStream(new FileOutputStream(jarFile), manifest);
```
Comment (tristantarrant) [inline jcache/embedded/src/test/java/org/infinispan/jcache/JCacheConfigurationTest.java]: For the test, the jar file should be created in the tmp directory. See org.infinispan.test.TestingUtil#tmpDirectory(java.lang.Class<?>).

Comment (dmurat) [reply jcache/embedded/src/test/java/org/infinispan/jcache/JCacheConfigurationTest.java]: Ok, now test tmp folder is used.

**Comment by tristantarrant** [comment]
Squashed and pushed to both master and 9.1.x. Thanks

**Comment by sammyhk** [comment]
Suspected the fix is not correct, considering uri `jar:file:/path/to/my/jar.jar!/infinispan.xml`, [AbstractFileLookup.java#L69](https://github.com/infinispan/infinispan/blob/5d3d2f1c0a3c41d60158d2b771d43abb3d4a5233/commons/src/main/java/org/infinispan/commons/util/AbstractFileLookup.java#L69) extracted `insideJarFilePath = "/infinispan.xml"`. As `getAsInputStreamFromClassLoader(...)` implementation use `ClassLoader.getResourceAsStream(...)` to lookup the resource, any resource starting with "/" actually will always return null.

**Pull request closed without merging**